.github/workflows/docker-dev-release.yml

```diff
@@ -115,17 +128,34 @@ jobs:
           registry: ghcr.io
           username: ${{ github.repository_owner }}
           password: ${{ secrets.GITHUB_TOKEN }}
-      
+
+      - name: Login to Gcp Artifact Registry
+        uses: docker/login-action@v3
+        with:
+          registry: us-central1-docker.pkg.dev
+          username: _json_key
+          password: ${{ secrets.GCP_SA_KEY }}
+
       - name: Merge and Push
         run: |
             flavor_tag=${{ env.image }}:${{matrix.flavor}}
             sha_tag=${flavor_tag}-${{ needs.build_and_tag.outputs.sha }}
-            
+
             # Create and push the manifest like dragonfly-dev:alpine-<sha>
             docker manifest create ${sha_tag} --amend ${sha_tag}-amd64 --amend ${sha_tag}-arm64
             docker manifest push ${sha_tag}
 
             # Create and push the manifest like dragonfly-dev:alpine
             docker manifest create ${flavor_tag} --amend ${sha_tag}-amd64 --amend ${sha_tag}-arm64
             docker manifest push ${flavor_tag}
-            
\ No newline at end of file
+
```

Comment:
seems like lots of code duplication. Can you introduce a bash function that runs docker manifest create and push?